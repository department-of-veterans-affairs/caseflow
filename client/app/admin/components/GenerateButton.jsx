/* eslint-disable func-style */
import React, { useEffect, useState } from 'react';
import Button from '../../components/Button';
import Modal from '../../components/Modal';
import Alert from '../../components/Alert';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from 'app/constants/AppConstants';
import PropTypes from 'prop-types';
import { CSVLink } from 'react-csv';

const GenerateButton = (props) => {

  // state properties
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [showEmptyResultsBanner, setShowEmptyResultsBanner] = useState(false);
  const [showErrorBanner, setShowErrorBanner] = useState(false);
  const [extractedResults, setExtractedResults] = useState('');

  useEffect(() => {
    setIsLoading(props.isLoading);
  }, [props.isLoading]);

  useEffect(() => {
    setExtractedResults(props.extractedResults);

    if (props.extractedResults?.length > 0) {
      setShowEmptyResultsBanner(false);
      setShowConfirmModal(true);
    }
  }, [props.extractedResults]);

  useEffect(() => {
    if (props.emptyResultsMessage?.length > 0) {
      setShowEmptyResultsBanner(true);
      setShowConfirmModal(false);
    }
  }, [props.emptyResultsMessage]);

  useEffect(() => {
    if (props.manualExtractionSuccess === false) {
      setShowErrorBanner(true);
    } else {
      setShowErrorBanner(false);
    }
  }, [props.manualExtractionSuccess]);

  const onClickGenerate = () => {
    props.sendExtractRequest();
  };

  return (
    <div style={{ height: '75vh' }}>
      {
        showEmptyResultsBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="No Veterans were found" type="success" />
        </div>
      }
      {
        showErrorBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="Veteran Extract Failed" type="error" />
        </div>
      }
      {
        !isLoading &&
        <Button
          id="generate-extract"
          onClick={() => onClickGenerate()}
        >
          Generate Veteran Extract
        </Button>
      }
      {
        showConfirmModal &&

        <Modal title="This file contains PII"
          confirmButton={<CSVLink data={extractedResults} filename="veteran_extract.csv" onClick={() => setShowConfirmModal(false)}>Confirm</CSVLink>}
          closeHandler={() => {
            setShowConfirmModal(false);
          }}
        >
          Are you sure you want to download it?
        </Modal>
      }
      {
        isLoading &&

        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Action is Running...
          </div>
        </LoadingContainer>
      }
    </div>
  );
};

GenerateButton.propTypes = {
  extractedResults: PropTypes.string,
  manualExtractionSuccess: PropTypes.bool,
  isLoading: PropTypes.bool,
  sendExtractRequest: PropTypes.func,
  emptyResultsMessage: PropTypes.string,
};

export default GenerateButton;
