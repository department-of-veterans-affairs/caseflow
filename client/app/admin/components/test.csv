VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN