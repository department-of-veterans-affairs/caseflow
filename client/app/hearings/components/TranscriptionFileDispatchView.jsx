import React, { useState } from 'react';
import QueueOrganizationDropdown from '../../queue/components/QueueOrganizationDropdown';
import AppSegment from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/AppSegment';
import { css } from 'glamor';
import TabWindow from '../../components/TabWindow';
import { tabConfig } from './TranscriptionFileDispatchTabs';

export const TranscriptionFileDispatchView = () => {
  const [selectedFiles, setSelectedFiles] = useState([]);

  const buildPackage = () => {
    // build the package
  };

  const selectFilesForPackage = (files) => {
    setSelectedFiles(files.filter((file) => file.status === 'selected'));
  };

  return (
    <AppSegment filledBackground >
      <h1 {...css({ display: 'inline-block' })}>Transcription file dispatch</h1>
      <QueueOrganizationDropdown organizations={[{ name: 'Transcription', url: 'transcription-team' }]} />
      <TabWindow
        name="transcription-tabwindow"
        defaultPage={0}
        fullPage={false}
        tabs={tabConfig(buildPackage, selectFilesForPackage, selectedFiles.length)}
      />
    </AppSegment>
  );
};
