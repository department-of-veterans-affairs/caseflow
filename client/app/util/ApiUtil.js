import request from 'superagent';
import nocache from 'superagent-no-cache';
import ReactOnRails from 'react-on-rails';

// TODO(jd): Fill in other HTTP methods as needed
const ApiUtil = {

  // Converts regular language to camelCase
  // 'VACOLS - 123, User' becomes 'vacolsUser'
  convertToCamelCase(phrase = '') {
    // Code courtesy of Stack Overflow, Question 2970525
    return phrase.toLowerCase().
        replace(/[^a-zA-Z ]/g, "").
        replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, (match, index) => {
          if (Number(match) === 0) {
            return "";
          }

          return index === 0 ? match.toLowerCase() : match.toUpperCase();
        });
  },

  // Converts camelCase to snake_case
  convertToSnakeCase(data = {}) {
    let result = {};

    for (let key in data) {
      if ({}.hasOwnProperty.call(data, key)) {
        // convert key from camelCase to snake_case
        let snakeKey = key.replace(/([A-Z])/g, ($1) => `_${$1.toLowerCase()}`);

        // assign value to new object
        result[snakeKey] = data[key];
      }
    }

    return result;
  },

  get(url, options = {}) {
    return request.
      get(url).
      set(this.headers(options.headers)).
      query(options.query).
      use(nocache);
  },

  // Default headers needed to talk with rails server.
  // Including rail's authenticity token
  headers(options = {}) {
    let headers = Object.assign({},
      {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      ReactOnRails.authenticityHeaders(),
      options);


    return headers;
  },

  patch(url, options = {}) {
    return request.
      post(url).
      set(this.headers({ 'X-HTTP-METHOD-OVERRIDE': 'patch' })).
      send(options.data).
      use(nocache);
  },

  post(url, options = {}) {
    return request.
      post(url).
      set(this.headers(options.headers)).
      send(options.data).
      use(nocache);
  }
};

export default ApiUtil;
