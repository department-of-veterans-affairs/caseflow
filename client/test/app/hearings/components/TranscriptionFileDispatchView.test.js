import React from 'react';
import { TranscriptionFileDispatchView } from '../../../../app/hearings/components/TranscriptionFileDispatchView';
import { render, waitFor, screen, cleanup } from '@testing-library/react';
import ApiUtil from '../../../../app/util/ApiUtil';
import { when } from 'jest-when';
import { axe } from 'jest-axe';
import userEvent from '@testing-library/user-event';
import { MemoryRouter as Router } from 'react-router-dom';

const setup = () => render( <Router><TranscriptionFileDispatchView /></Router>);

const mockTranscriptionFiles = [
  {
    id: 40,
    externalAppealId: 'b5eba21a-9baf-41a3-ac1c-08470c2b79c4',
    docketNumber: '200103-61110',
    caseDetails: 'Danial Reynolds (000543695)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '11/02/2020',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 39,
    externalAppealId: '12bb84ff-65fb-4422-bee4-fe7553fdf5c3',
    docketNumber: '190227-4821',
    caseDetails: 'Craig Wintheiser (000562812)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '08/27/2020',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 38,
    externalAppealId: '12bb84ff-65fb-4422-bee4-fe7553fdf5c3',
    docketNumber: '190227-4821',
    caseDetails: 'Craig Wintheiser (000562812)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '04/30/2020',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 37,
    externalAppealId: '12bb84ff-65fb-4422-bee4-fe7553fdf5c3',
    docketNumber: '190227-4821',
    caseDetails: 'Craig Wintheiser (000562812)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '01/13/2020',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 36,
    externalAppealId: '12bb84ff-65fb-4422-bee4-fe7553fdf5c3',
    docketNumber: '190227-4821',
    caseDetails: 'Craig Wintheiser (000562812)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '10/24/2019',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 35,
    externalAppealId: '9c14a2fd-348e-44d0-9465-5c3c6303b52d',
    docketNumber: '180910-667',
    caseDetails: 'Bud Hessel (000654829)',
    isAdvancedOnDocket: true,
    caseType: 'Original',
    hearingDate: '04/04/2019',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 34,
    externalAppealId: '3ad0bcc7-613f-4fed-8088-024c93b2cb86',
    docketNumber: '230828-1803',
    caseDetails: 'Vernon Weimann (300000588)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '07/01/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 33,
    externalAppealId: '572480c5-f621-4038-84ae-fe96c8012f80',
    docketNumber: '230829-1801',
    caseDetails: 'Kirby Howe (300000587)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '07/01/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 32,
    externalAppealId: 'fbad9355-9bef-4957-92a3-f88e467aad66',
    docketNumber: '230830-1799',
    caseDetails: 'Glendora Parisian (300000583)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '06/18/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 31,
    externalAppealId: '6ec0df25-a7da-4a3d-8633-4120565ec8a6',
    docketNumber: '230831-1797',
    caseDetails: 'Buford Wunsch (300000582)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '06/18/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 30,
    externalAppealId: 'fca24f0c-30b8-4a41-b077-d0d140d6b266',
    docketNumber: '230901-1795',
    caseDetails: 'Renae Hansen (300000581)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '06/18/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 29,
    externalAppealId: '7ba4da54-6ed8-49c5-b79d-884539742990',
    docketNumber: '230902-1793',
    caseDetails: 'Donovan Doyle (300000578)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '07/01/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 28,
    externalAppealId: 'eeb095a4-3e43-4c4e-9f17-4cfd0d2af751',
    docketNumber: '230903-1791',
    caseDetails: 'Darrin Heathcote (300000577)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '07/01/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 27,
    externalAppealId: '040e56c6-5834-447e-b6d5-0f9a1d2007d3',
    docketNumber: '230904-1789',
    caseDetails: 'Timothy Spencer (300000573)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '06/18/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  },
  {
    id: 26,
    externalAppealId: '7b3f97c3-af0c-4b6c-9ac1-566382a66565',
    docketNumber: '230905-1787',
    caseDetails: 'Evia Gerhold (300000572)',
    isAdvancedOnDocket: false,
    caseType: 'Original',
    hearingDate: '06/18/2024',
    hearingType: 'Hearing',
    fileStatus: 'Successful upload (AWS)'
  }
];

const mockTranscriptionFilesResponse = {
  body: {
    task_page_count: 3,
    tasks: {
      data: mockTranscriptionFiles
    },
    tasks_per_page: 15,
    total_task_count: 40
  }
};

const mockLockedResponse = {
  body: [
    {
      id: 40,
      status: 'selected',
      message: ''
    },
    {
      id: 39,
      status: 'locked',
      message: 'Locked by QATTY2'
    }
  ]
};

const mockClickedResponse = {
  body: [
    {
      id: 40,
      status: 'selected',
      message: ''
    },
    {
      id: 39,
      status: 'locked',
      message: 'Locked by QATTY2'
    },
    {
      id: 38,
      status: 'selected',
      message: ''
    }
  ]
};

const constClickData = { data: { file_ids: [37], status: true } };

describe('TranscriptionFileDispatch', () => {
  beforeEach(async () => {
    ApiUtil.get = jest.fn();
    ApiUtil.post = jest.fn();

    when(ApiUtil.get).calledWith('/hearings/transcription_files/transcription_file_tasks?tab=Unassigned&page=1').
      mockResolvedValueOnce(mockTranscriptionFilesResponse);

    when(ApiUtil.get).calledWith('/hearings/transcription_files/locked').
      mockResolvedValueOnce(mockLockedResponse);

    when(ApiUtil.post).calledWith('/hearings/transcription_files/lock', constClickData).
      mockResolvedValueOnce(mockClickedResponse);
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it('passes a11y', async () => {
    const { container } = setup();
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('has switch views dropdown', () => {
    setup();

    expect(screen.getByText('Switch views')).toBeInTheDocument();
  });

  it('has the correct tabs', () => {
    const { container } = setup();
    const tabs = container.querySelectorAll('.cf-tab');

    expect(tabs).toHaveLength(4);
    expect(tabs[0].textContent).toBe('Unassigned');
    expect(tabs[1].textContent).toBe('Assigned');
    expect(tabs[2].textContent).toBe('Completed');
    expect(tabs[3].textContent).toBe('All transcription');
  });

  describe('Unassigned Tab', () => {

    it('loads a table from backend data and handles selected and locked records', async () => {
      const { container } = setup();

      await waitFor(() =>
        expect(screen.getAllByText('Viewing 1-15 of 40 total')[0]).toBeInTheDocument()
      );

      expect(screen.getByText('Another user is in the assignment queue.')).toBeInTheDocument();
      expect(screen.getByText('1 file selected')).toBeInTheDocument();

      const select = container.querySelectorAll('.select-file');
      const checkboxes = container.querySelectorAll('.select-file input');

      expect(checkboxes[0]).toBeChecked();

      expect(checkboxes[1]).toBeDisabled();
      expect(select[1]).toHaveAttribute('title', 'Locked by QATTY2');

      expect(container).toMatchSnapshot();
    });

    it('allows a user to click to lock a record and call the back end', async () => {
      const { container } = setup();

      await waitFor(() =>
        expect(screen.getAllByText('Viewing 1-15 of 40 total')[0]).toBeInTheDocument()
      );

      const checkboxes = container.querySelectorAll('.select-file input');

      expect(checkboxes[3]).not.toBeChecked();

      userEvent.click(checkboxes[3]);

      expect(checkboxes[3]).toBeChecked();
      expect(ApiUtil.post).toHaveBeenCalledWith(
        '/hearings/transcription_files/lock', constClickData);
    });
  });
});
