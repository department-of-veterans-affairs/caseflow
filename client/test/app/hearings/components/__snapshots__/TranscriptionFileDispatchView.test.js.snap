// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranscriptionFileDispatch matches snapshot 1`] = `
<div>
  <section
    class="cf-app-segment cf-app-segment--alt"
  >
    <h1
      data-css-w0yocm=""
    >
      Transcription file dispatch
    </h1>
    <div
      class="cf-dropdown"
      data-css-7w6cer=""
    >
      <div
        class="cf-dropdown"
        data-css-fp8ian=""
      >
        <button
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Switch views"
          aria-pressed="false"
          class="cf-dropdown-trigger usa-button usa-button-secondary"
          data-css-x1q03x=""
          role="button"
        >
          Switch views
        </button>
      </div>
    </div>
    <div
      class="cf-tab-navigation"
      role="tablist"
    >
      <button
        aria-controls="transcription-tabwindow-tabpanel-0"
        aria-selected="true"
        class="cf-tab tab cf-active"
        data-value="0"
        id="transcription-tabwindow-tab-0"
        role="tab"
        tabindex="0"
      >
        <div>
          <span>
            
            <span>
              Unassigned
            </span>
            
          </span>
        </div>
      </button>
      <button
        aria-controls="transcription-tabwindow-tabpanel-1"
        aria-selected="false"
        class="cf-tab tab"
        data-value="1"
        id="transcription-tabwindow-tab-1"
        role="tab"
        tabindex="-1"
      >
        <div>
          <span>
            
            <span>
              Assigned
            </span>
            
          </span>
        </div>
      </button>
      <button
        aria-controls="transcription-tabwindow-tabpanel-2"
        aria-selected="false"
        class="cf-tab tab"
        data-value="2"
        id="transcription-tabwindow-tab-2"
        role="tab"
        tabindex="-1"
      >
        <div>
          <span>
            
            <span>
              Completed
            </span>
            
          </span>
        </div>
      </button>
      <button
        aria-controls="transcription-tabwindow-tabpanel-3"
        aria-selected="false"
        class="cf-tab tab"
        data-value="3"
        id="transcription-tabwindow-tab-3"
        role="tab"
        tabindex="-1"
      >
        <div>
          <span>
            
            <span>
              All transcription
            </span>
            
          </span>
        </div>
      </button>
    </div>
    <div
      class="cf-tab-window-body-full-screen tabContent"
    >
      <div
        aria-hidden="false"
        class="tabPanel active"
        id="transcription-tabwindow-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tab-description"
          style="display: flex; justify-content: space-between; margin-bottom: 2em; margin-top: -0.1em;"
        >
          Transcription owned by the Transcription Team are unassigned to a contractor:
          <a>
            <span
              style="display: inline-flex; font-size: small; cursor: pointer;"
            >
              Transcription settings
              <svg
                class=""
                height="17"
                version="1.1"
                viewBox="0 0 18 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="#0071bc"
                    fill-rule="nonzero"
                  >
                    <g
                      transform="translate(1.000000, 1.000000)"
                    >
                      <path
                        d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                        id="external-link-outline-box"
                      />
                      <path
                        d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                        id="external-link-outline-arrow"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </span>
          </a>
        </div>
        <div
          class="cf-search-ahead-parent"
          style="display: flex; justify-content: space-between; margin-bottom: 2em; margin-top: 3em;"
        >
          Please select the files you would like to dispatch for transcription:
          <div
            data-css-1n3u423=""
          >
            <span
              class="usa-search usa-search-big cf-search-ahead"
              role="search"
            >
              <label
                class="usa-search-big"
                for="transcription-table-search"
              >
                Search by Docket Number, Claimant Name, File Number, or SSN
              </label>
              <input
                aria-label="Enter your search term"
                class="cf-search-input-with-close"
                id="transcription-table-search"
                name="search"
                placeholder="Type to search..."
                type="search"
                value=""
              />
            </span>
          </div>
        </div>
        <div
          class="file-select"
          style="margin-top: -2em;"
        >
          <h2>
            0 files selected
          </h2>
          <div
            class="button-row"
            style="display: inline-block;"
          >
            <span>
              <button
                class="cf-submit usa-button-disabled usa-button"
                disabled=""
                type="button"
              >
                Package files
              </button>
            </span>
            <span>
              <button
                class="cf-submit cf-btn-link usa-button"
                type="button"
              >
                Cancel
              </button>
            </span>
          </div>
        </div>
        <div
          style="margin-top: 1em;"
        >
          <div
            data-css-1bh5zas=""
          >
            <div
              class="cf-table-wrapper"
            >
              <section
                class="cf-app-segment cf-app-segment--alt"
              >
                <div
                  data-css-1ogvp9u=""
                >
                  <div>
                    
                    <div
                      class="cf-loading-button-symbol cf-small-loader-symbol"
                    >
                      <svg
                        class="cf-react-icon-loading-back"
                        height="150px"
                        viewBox="0 0 500 500"
                        width="150px"
                      >
                        <path
                          d="M249.9,362c-6.5,0-12.8,1.2-18.9,3.5l-34.4,13.3l12.7,28.8l33-12.8c2.4-0.9,5-1.4,7.5-1.4s5.1,0.5,7.5,1.4 l33.2,12.8l12.7-28.8l-34.6-13.4C262.7,363.2,256.4,362,249.9,362 M418.9,319.9l-28.8,12.7l19.1,49.6c3.5,9,0.1,15.9-2.1,19.3 c-4,5.8-10.4,9.3-17.2,9.3h0c-2.6,0-5.2-0.5-7.7-1.5l-49.5-19.1L320,418.9l50.9,19.7c6.2,2.4,12.6,3.6,19,3.6h0 c17.1,0,33.2-8.6,43.1-23c9.8-14.3,11.9-32,5.5-48.3L418.9,319.9 M81,319.9l-19.7,51c-6.3,16.4-4.3,34,5.5,48.3 c9.9,14.4,26,23,43.1,23c6.4,0,12.9-1.2,19-3.6l51-19.7l-12.7-28.8l-49.5,19.1c-2.5,1-5.1,1.4-7.6,1.4c-6.8,0-13.3-3.4-17.3-9.3 c-2.3-3.3-5.6-10.3-2.1-19.2l19.1-49.6L81,319.9 M121.1,196.6l-28.8,12.7l12.8,33.1c1.9,4.9,1.9,10.2,0,15.1l-12.8,33.1l28.8,12.7 l13.3-34.5c4.7-12.2,4.7-25.6,0-37.7L121.1,196.6 M378.8,196.5l-13.3,34.5c-4.7,12.2-4.7,25.6,0,37.7l13.3,34.5l28.8-12.7 l-12.8-33.1c-1.9-4.9-1.9-10.2,0-15.1l12.8-33.1L378.8,196.5 M290.6,92.3l-33.2,12.8c-2.4,0.9-5,1.4-7.5,1.4 c-2.6,0-5.1-0.5-7.5-1.4l-33-12.7l-12.7,28.8l34.4,13.3c6,2.3,12.4,3.5,18.8,3.5c6.4,0,12.8-1.2,18.9-3.5l34.6-13.4L290.6,92.3 M110,57.8c-17.1,0-33.2,8.6-43.1,23c-9.8,14.3-11.9,32-5.5,48.3L81,180l28.8-12.7l-19.1-49.4c-3.5-9-0.1-15.9,2.1-19.2 c4-5.9,10.5-9.3,17.3-9.3c2.5,0,5.1,0.5,7.6,1.4l49.5,19.1L180,81.1l-50.9-19.7C122.9,59,116.5,57.8,110,57.8 M389.9,57.8 c-6.4,0-12.9,1.2-19,3.6L319.9,81l12.7,28.8l49.5-19.1c2.5-1,5.1-1.4,7.6-1.4c6.8,0,13.3,3.4,17.3,9.3c2.3,3.3,5.6,10.3,2.1,19.2 l-19.1,49.5l28.8,12.7l19.7-50.9c6.3-16.4,4.3-34-5.5-48.3C423.1,66.3,407,57.8,389.9,57.8M303.4,378.9l-12.7,28.8l29.3,11.3l12.7-28.8L303.4,378.9 M196.6,378.8l-29.3,11.3l12.7,28.8l29.3-11.3 L196.6,378.8 M407.6,290.6l-28.8,12.7l11.3,29.3l28.8-12.7L407.6,290.6 M92.4,290.5L81,319.9l28.8,12.7l11.3-29.3L92.4,290.5 M109.8,167.2L81,180l11.3,29.3l28.8-12.7L109.8,167.2 M390.1,167.2l-11.3,29.3l28.8,12.7l11.3-29.3L390.1,167.2 M180,81.1 l-12.7,28.8l29.3,11.3l12.7-28.8L180,81.1 M319.9,81l-29.3,11.3l12.7,28.8l29.3-11.3L319.9,81"
                          fill="#11598D"
                          fill-opacity="1"
                          opacity="1"
                        />
                      </svg>
                      <svg
                        class="cf-react-icon-loading-front"
                        height="150px"
                        style="margin-left: -150px;"
                        viewBox="0 0 500 500"
                        width="150px"
                      >
                        <path
                          d="M209.3,407.5L180,418.9l22.1,50C210.6,488.1,229,500,250,500s39.4-11.9,47.9-31.1l22.1-49.9l-29.3-11.3 l-21.5,48.5c-3.5,7.8-10.6,12.5-19.1,12.5c-8.5,0-15.7-4.7-19.1-12.5L209.3,407.5 M378.8,303.3l-33.8,15 c-11.9,5.3-21.4,14.7-26.7,26.7l-15,33.9l29.3,11.3l14.4-32.5c2.1-4.8,5.9-8.6,10.7-10.7l32.4-14.3L378.8,303.3 M121.1,303.3 l-11.3,29.3l32.4,14.4c4.8,2.1,8.6,5.9,10.7,10.7l14.4,32.5l29.3-11.3l-15-33.9c-5.3-11.9-14.8-21.4-26.7-26.7L121.1,303.3 M81,180 l-49.9,22.1C11.9,210.6,0,228.9,0,249.9c0,21,11.9,39.4,31.1,47.9L81,319.9l11.3-29.3l-48.5-21.5c-7.8-3.5-12.5-10.6-12.5-19.1 c0-8.5,4.7-15.7,12.5-19.1l48.5-21.5L81,180 M418.9,179.9l-11.3,29.3l48.6,21.5c7.8,3.5,12.5,10.6,12.5,19.1 c0,8.5-4.7,15.7-12.5,19.1l-48.6,21.5l11.3,29.3l50-22.1c19.2-8.5,31.1-26.9,31.1-47.9s-11.9-39.4-31.1-47.9L418.9,179.9 M167.3,109.8L153,142.2c-2.1,4.8-5.9,8.6-10.7,10.7l-32.5,14.4l11.3,29.3l33.9-15c11.9-5.3,21.4-14.7,26.7-26.7l14.9-33.7 L167.3,109.8 M332.7,109.8l-29.3,11.3l15,33.8c5.3,11.9,14.7,21.4,26.7,26.7l33.8,15l11.3-29.3l-32.4-14.4 c-4.8-2.1-8.6-5.9-10.7-10.7L332.7,109.8 M250,0c-21,0-39.4,11.9-47.9,31.2L180,81.1l29.3,11.3l21.5-48.5 c3.5-7.8,10.6-12.5,19.1-12.5c8.5,0,15.7,4.7,19.1,12.5l21.5,48.5L319.9,81l-22.1-49.9C289.3,11.9,271,0,250,0M303.4,378.9l-12.7,28.8l29.3,11.3l12.7-28.8L303.4,378.9 M196.6,378.8l-29.3,11.3l12.7,28.8l29.3-11.3 L196.6,378.8 M407.6,290.6l-28.8,12.7l11.3,29.3l28.8-12.7L407.6,290.6 M92.4,290.5L81,319.9l28.8,12.7l11.3-29.3L92.4,290.5 M109.8,167.2L81,180l11.3,29.3l28.8-12.7L109.8,167.2 M390.1,167.2l-11.3,29.3l28.8,12.7l11.3-29.3L390.1,167.2 M180,81.1 l-12.7,28.8l29.3,11.3l12.7-28.8L180,81.1 M319.9,81l-29.3,11.3l12.7,28.8l29.3-11.3L319.9,81"
                          fill="#323a45"
                          fill-opacity=".25"
                          opacity="1"
                        />
                      </svg>
                    </div>
                  </div>
                  <p />
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="tabPanel"
        id="transcription-tabwindow-tabpanel-1"
        role="tabpanel"
      />
      <div
        aria-hidden="true"
        class="tabPanel"
        id="transcription-tabwindow-tabpanel-2"
        role="tabpanel"
      />
      <div
        aria-hidden="true"
        class="tabPanel"
        id="transcription-tabwindow-tabpanel-3"
        role="tabpanel"
      />
    </div>
  </section>
</div>
`;
